rdr_service/resource/generators/pdr_participant.py

```diff
@@ -0,0 +1,223 @@
+#
+# This file is subject to the terms and conditions defined in the
+# file 'LICENSE', which is part of this source code package.
+#
+import os
+
+from rdr_service.dao.resource_dao import ResourceDataDao
+from rdr_service.model.bq_base import BQRecord
+from rdr_service.resource import generators, schemas
+from rdr_service.resource.generators.participant import ParticipantSummaryGenerator
+from rdr_service.resource.schemas.participant import StreetAddressTypeEnum
+
+
+class PDRParticipantSummaryGenerator(generators.BaseGenerator):
+    """
+    Generate a Participant Summary Resource object
+    """
+    ro_dao = None
+    rural_zipcodes = None
+
+    def make_resource(self, p_id, ps_res=None):
+        """
+        Build a Participant Summary BQRecord object for the given participant id.
+        :param p_id: participant id
+        :param ps_res: A BQParticipantSummary BQRecord object.
+        :return: BQRecord object
+        """
+        if not self.ro_dao:
+            self.ro_dao = ResourceDataDao(backup=True)
+
+        # Since we are primarily a subset of the Participant Summary, call the full Participant Summary generator
+        # and take what we need from it.
+        if not ps_res:
+            ps_res = ParticipantSummaryGenerator().make_resource(p_id)
+        res = BQRecord(schema=None, data=ps_res.get_data())
+
+        if hasattr(res, 'addresses') and res.addresses:
+            for address in res.addresses:
+                if address['addr_type_id'] == StreetAddressTypeEnum.RESIDENCE.value:
+                    setattr(res, 'addr_state', address['addr_state'])
+                    if address['addr_zip']:
+                        setattr(res, 'addr_zip', address['addr_zip'][:3])
+
+        summary = res.to_dict()
+        # Summarize biobank order data if there are biobank orders.
+        if hasattr(res, 'biobank_orders'):
```

Comment:
As discussed offline for handling fixes for PDR-243, this will likely need updating to correct issues with OrderStatus enum values (COLLECTED, PROCESSED, FINALIZED) vs. BiobankOrderStatus enum values (CANCELLED, AMENDED)